### PR TITLE
Ceds 1860 notifications journey

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

File: app/controllers/ChoiceController.scala [hunk 1]
```diff
@@ -80,6 +80,8 @@ class ChoiceController @Inject()(
             proceedJourney(DisassociateUcrAnswers(), controllers.consolidations.routes.DisassociateDucrController.display())
           case forms.Choice.ShutMUCR =>
             proceedJourney(ShutMucrAnswers(None), routes.ChoiceController.displayPage())
+          case forms.Choice.ViewSubmissions =>
+            proceedJourney(ViewSubmissionsAnswers(), routes.ChoiceController.displayPage())
```
Comment (PatrykRudnicki) [inline app/controllers/ChoiceController.scala]: You shouldn't use `proceedJourney` for submissions. This method save basic model in Mongo

File: app/models/cache/JourneyType.scala [hunk 2]
```diff
@@ -21,7 +21,7 @@ import play.api.libs.json.{Format, Reads, Writes}
 object JourneyType extends Enumeration {
   type JourneyType = Value
 
-  val ARRIVE, DEPART, ASSOCIATE_UCR, DISSOCIATE_UCR, SHUT_MUCR = Value
+  val ARRIVE, DEPART, ASSOCIATE_UCR, DISSOCIATE_UCR, SHUT_MUCR, VIEW_SUBMISSIONS = Value
```
Comment (PatrykRudnicki) [inline app/models/cache/JourneyType.scala]: I'm not sure about approach, I think we should handle submission in slightly different way, without Answers, without JourneyType, just Choice

File: app/controllers/ChoiceController.scala [hunk 3]
```diff
@@ -58,9 +58,9 @@ class ChoiceController @Inject()(
       case forms.Choice.DisassociateUCR =>
         proceedJourney(DisassociateUcrAnswers(), controllers.consolidations.routes.DisassociateDucrController.display())
       case forms.Choice.ShutMUCR =>
-        proceedJourney(AssociateUcrAnswers(), routes.ChoiceController.displayPage())
-      case forms.Choice.Submissions =>
         proceedJourney(ShutMucrAnswers(), routes.ChoiceController.displayPage())
+      case forms.Choice.ViewSubmissions =>
+        proceedJourney(ViewSubmissionsAnswers(), routes.ChoiceController.displayPage())
```
Comment (ed0906) [inline app/controllers/ChoiceController.scala]: We dont need answers for this journey

File: app/controllers/ChoiceController.scala (hunk 1 above)
Comment (ed0906) [inline app/controllers/ChoiceController.scala]: Similar, we dont need Answers for this journey, just redirect to the relevant page

File: app/forms/Choice.scala [hunk 4]
```diff
@@ -37,11 +36,12 @@ object Choice {
...
-    case DEPART         => Departure
-    case ASSOCIATE_UCR  => AssociateUCR
-    case DISSOCIATE_UCR => DisassociateUCR
-    case SHUT_MUCR      => ShutMUCR
+    case ARRIVE           => Arrival
+    case DEPART           => Departure
+    case ASSOCIATE_UCR    => AssociateUCR
+    case DISSOCIATE_UCR   => DisassociateUCR
+    case SHUT_MUCR        => ShutMUCR
+    case VIEW_SUBMISSIONS => ViewSubmissions
```
Comment (ed0906) [inline app/forms/Choice.scala]: No need for this line

File: app/models/cache/Answers.scala [hunk 5]
```diff
@@ -71,13 +71,21 @@ object DisassociateUcrAnswers {
...
 case class ShutMucrAnswers(override val eori: Option[String] = None) extends Answers {
-  override val `type`: JourneyType.Value = JourneyType.DISSOCIATE_UCR
+  override val `type`: JourneyType.Value = JourneyType.SHUT_MUCR
 }
 
 object ShutMucrAnswers {
   implicit val format: Format[ShutMucrAnswers] = Json.format[ShutMucrAnswers]
 }
 
+case class ViewSubmissionsAnswers(override val eori: Option[String] = None) extends Answers {
```
Comment (ed0906) [inline app/models/cache/Answers.scala]: As mentioned above, this journey doesn't need user answers

File: test/controllers/ChoiceControllerSpec.scala [hunk 6]
```diff
@@ -129,11 +154,12 @@ class ChoiceControllerSpec extends ControllerLayerSpec with MockCache {
...
   "POST" should {
 
-    implicit val postRequest = FakeRequest("POST", "/").withCSRFToken
+    def postWithChoice(choice: Choice): Request[AnyContentAsFormUrlEncoded] =
+      FakeRequest("POST", "/").withFormUrlEncodedBody("choice" -> choice.value).withCSRFToken
 
     "return 303 (SEE_OTHER) when authenticated" when {
 
-      "arrival" in {
+      "user choose arrival" in {
```
Comment (ed0906) [inline test/controllers/ChoiceControllerSpec.scala]: `chooses`

Comment (MaciejRewera) [reply test/controllers/ChoiceControllerSpec.scala]: 👍

File: test/controllers/ChoiceControllerSpec.scala [hunk 7]
```diff
@@ -169,27 +195,34 @@ class ChoiceControllerSpec extends ControllerLayerSpec with MockCache {
...
+      val result = controller().submit(FakeRequest("POST", "/").withCSRFToken)
 
       status(result) mustBe BAD_REQUEST
-      contentAsHtml(result) mustBe page(Choice.form().bind(Map[String, String]()))
     }
 
     "return 403 when unauthenticated" in {
 
-      val result = controller(UnsuccessfulAuth).submit(postRequest)
+      val result = controller(UnsuccessfulAuth).submit(FakeRequest("POST", "/").withCSRFToken)
```
Comment (PatrykRudnicki) [inline test/controllers/ChoiceControllerSpec.scala]: Please extract this `FakeRequest` to ControllerLayerSpec, so we will be able to use `postRequest` as empty POST and `postRequest(json)` as post with json

**Pull request merged**